ramls/check-in-by-barcode-response.json

```diff
@@ -77,10 +77,18 @@
           "properties": {
             "name": {
               "description": "Name of the item status",
-              "type": "string"
+              "type": "string",
+              "$ref": "schema/item-status-names.json"
```

Comment:
What is the purpose of an enumeration here?

I think we need to be really cautious with this approach. What happens when a new item status is introduced in inventory? Can that new status be presented here?

Reply:
Well, this enum does not restrict anything, in a fact (because CIRC is not an RMB module, so there is no validation against schema). 

However, there was restriction before those changes. We've being used `ItemStatus.java` enum for item statuses, with defined elements.

Reply:
> Well, this enum does not restrict anything, in a fact (because CIRC is not an RMB module, so there is no validation against schema).

Whilst no representation is validated against this schema (yet), this schema forms part of a contract with the client.

If inventory introduces the `Recently returned` status and an item with that status is included in a check in response, what should the client do, given that the response is invalid according to the schema?

(RAML Module Builder also does not validate against schema for outgoing representations, only incoming).
 
> However, there was restriction before those changes. We've being used `ItemStatus.java` enum for item statuses, with defined elements.

That restriction is internal to the module, it is the server's choice that it only uses a certain set of statuses. The client is ignorant of this.

Reply:
> If inventory introduces the Recently returned status and an item with that status is included in a check in response, what should the client do, given that the response is invalid according to the schema?

I expect that endpoint will require updates for this case in order to handle this item at all. We use the `ItemStatus` enum which do not contain this new value, and because of this the item won't be correctly processed I think.

Reply:
> > If inventory introduces the Recently returned status and an item with that status is included in a check in response, what should the client do, given that the response is invalid according to the schema?
> 
> I expect that endpoint will require updates for this case in order to handle this item at all. 

You expect that the `circulation` interface has to change every time the `item-storage` interface changes the list of statuses?

> We use the `ItemStatus` enum which do not contain this new value, and because of this the item won't be correctly processed I think.

Does check in fail if an item is in an unknown status prior to check in? Do all circulation operations fail if the status is not understood?

Reply:
I've double checked the logic and it will be possible to make check in for an item with a status that are not present in circulation's enum, because check in operation always puts the item into available/in transit/awaiting pickup/delivery.

I'm not sure about another circulation operations.